src/legacy/ui/public/vis/__tests__/_agg_configs.js

```diff
@@ -353,5 +353,55 @@ describe('AggConfigs', function () {
         }
       }(topLevelDsl));
     });
+
+    it('adds the parent aggs of nested metrics at every level if the vis is hierarchical', function () {
+      const vis = new Vis(indexPattern, {
+        type: 'histogram',
+        aggs: [
+          {
+            id: '1',
+            type: 'avg_bucket',
+            schema: 'metric',
+            params: {
+              customBucket: {
+                id: '1-bucket',
+                type: 'date_histogram',
+                schema: 'bucketAgg',
+                params: {
+                  field: '@timestamp'
+                }
+              },
+              customMetric: {
+                id: '1-metric',
+                type: 'count',
+                schema: 'metricAgg',
+                params: {}
+              }
+            }
+          },
+          {
+            id: '2',
+            type: 'terms',
+            schema: 'bucket',
+            params: {
+              field: 'geo.src',
+            }
+          },
+          {
+            id: '3',
+            type: 'terms',
+            schema: 'bucket',
+            params: {
+              field: 'machine.os',
+            }
+          }
+        ]
+      });
+      vis.isHierarchical = _.constant(true);
+
+      const topLevelDsl = vis.aggs.toDsl(vis.isHierarchical())['2'];
+      expect(topLevelDsl.aggs).to.have.keys(['1', '1-bucket']);
```

Comment:
nit: It might be nice to include an assertion for the correct `buckets_path` as well; something like:

```javascript
expect(topLevelDsl.aggs['1'].avg_bucket).to.have.property('buckets_path', '1-bucket>_count');
```

I realize that isn't validating the precise problem you are fixing here, but for those looking at this test in the future, it could make it clearer how everything fits together & why we need to check that the parent aggs have been added at each level.